src/server/pps/server/poller.go

```diff
@@ -258,18 +278,23 @@ func (m *ppsMaster) watchPipelines(ctx context.Context) {
 			if event.Err != nil {
 				return errors.Wrapf(event.Err, "event err")
 			}
+			parts := strings.SplitN(string(event.Key), "@", 2)
```

Comment:
This should be a call to a helper function that is also called by the pipeline key check.